content/articles/building-scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/index.md

```diff
@@ -0,0 +1,115 @@
+# Multi-Module Project Setup in Spring Boot
+
+A three-man organisation in 2019 grew to become a twelve-man team fourteen months later. The growth in team members was necessitated by the growth in the project. However, for them also came an increase in the cost of building, maintaining and scaling the project; they were running the microservices architecture.
+
+In this tutorial, we will be going through the multi-module architecture - which is an drastic improvement on the monolith approach, yet offering as much possibilities of the microservices architecture, while evading the huge demands of microservices. We will be building a model of a hospital management system, taking into account some of the departments that make for a successful medical application.
+
+## Table of contents
+The tutorial will cover:
+- Why is the multi-module approach better than the monolith
+- Advantages of multi-module over microservices architecture
+- Setting a Spring Boot multi-module project
+- How can we make them interact with each other?
+- Gaining the benefits of microservices architecture from the multi-module architecture
+- Bonus
+- Conclusion
+
+## Prerequisites
+To make the most of this tutorial, it is required to have:
+•	Basic understanding of microservices.
+•	Familiarity with the Spring Boot framework.
+•	Good internet connection
+•	Intellij code editor installed.
+
+### Why is the multi-module better than monolith?
+Traditional, monolith design puts all the services for a project in one singular module. Everything is housed in one big box just as they are. In multi-module, each service is compartmentalised before being placed in the big box; the project. Multi-module allows for code re-usability unlike monolith. Code maintenance and adding new features or functionality is much easier.
+
+### Advantages of Multi-module over Microservices
+No doubt, the microservices architecture is beneficial. However, in comparison with the multi-module architecture, it appears to have some short-comings. The first would be ensuring availability of services at optimal time. Because services in the microservices architecture are deployed separately, it means that a downtime in one service could cripple an entire organisation, depending on how many services depend on it. Also, passing around dependencies can be tricky. For large teams and projects, this can be challenging. With the multi-module architecture, services are still developed individually but within a mother project that caters to all of them. Therefore, interest is not necessarily spread across several services but is channelled to the mother project and by extension, the child services. As a result, multi-module projects are easier to maintain than microservices.
```

Comment:
Break down this paragraph into smaller paras.

Reply:
Done. Thanks